libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/drillDownUtil.ts

```diff
@@ -0,0 +1,165 @@
+            }
+            return bucketItem;
+        },
+    );
+}
+
+function removePropertiesForRemovedAttributes(insight: IInsight): IInsight {
+    const properties: VisualizationProperties = insightProperties(insight);
+
+    if (!properties) {
+        return insight;
+    }
+
+    const identifiers = insightItems(insight).map((bucketItem: IAttributeOrMeasure) =>
+        bucketItemLocalId(bucketItem),
+    );
+
+    const result = Object.entries(properties).reduce((acc, [key, value]) => {
+        if (key === ENUM_PROPERTIES_TYPE.CONTROLS && value.columnWidths) {
+            const columns = value.columnWidths.filter(
+                (c: TColumnWidths) =>
+                    c?.attributeColumnWidthItem?.attributeIdentifier === undefined ||
+                    identifiers.includes(c.attributeColumnWidthItem.attributeIdentifier),
+            );
+
+            return {
+                ...acc,
+                [key]: {
+                    columnWidths: columns,
+                },
+            };
+        }
+
+        return { ...acc };
+    }, properties);
+
+    return insightSetProperties(insight, result);
+}
+
+export function sanitizeTableProperties(insight: IInsight): IInsight {
+    return removePropertiesForRemovedAttributes(insight);
+}
+
+export function convertIntersectionToFilters(intersections: IDrillEventIntersectionElement[]): IFilter[] {
+    return intersections
+        .map((intersection) => intersection.header)
+        .filter(isDrillIntersectionAttributeItem)
+        .map((header) =>
+            newPositiveAttributeFilter(uriRef(header.attributeHeader.uri), {
+                uris: [header.attributeHeaderItem.uri],
+            }),
+        );
+}
+
+export function addIntersectionFiltersToInsight(
+    source: IInsight,
+    intersection: IDrillEventIntersectionElement[],
+): IInsight {
+    const filters = convertIntersectionToFilters(intersection);
+    const resultFilters = [...source.insight.filters, ...filters];
+
+    return insightSetFilters(source, resultFilters);
+}
+
+export function getIntersectionPartAfter(
+    intersection: IDrillEventIntersectionElement[],
+    localIdentifier: string,
+): IDrillEventIntersectionElement[] {
+    const index = intersection.findIndex(
+        (item: IDrillEventIntersectionElement) =>
+            isDrillIntersectionAttributeItem(item.header) &&
+            item.header.attributeHeader.localIdentifier === localIdentifier,
+    );
+
+    return intersection.slice(index);
+}
+
+export function adjustIntersectionForColumnBar(
+    source: IInsight,
+    event: IDrillEvent,
+): IDrillEventIntersectionElement[] {
+    const hasStackByAttributes = bucketsFind(insightBuckets(source), (bucket: IBucket) => {
+        return idMatchBucket(BucketNames.STACK) && !bucketIsEmpty(bucket);
+    });
+
+    const intersection = event.drillContext.intersection;
+    return hasStackByAttributes ? arrayUtils.shiftArrayRight(intersection) : intersection;
+}
```

Comment:
perhaps move closer to the implementation?

Reply:
Moved